### PR TITLE
Certificates

### PR DESCRIPTION
**Pull request by nwinter**
Add course completion certificates that teachers and students can access and print for each course they've completed.
screenshot 2018-05-15 12 07 54
screenshot 2018-05-15 12 08 17
screenshot 2018-05-15 12 09 13

### CODE REVIEW

File: app/core/initialize.coffee [hunk 1]
```diff
@@ -54,21 +54,24 @@ handleNormalUrls = ->
...
+      return
 
-      # Remove leading slashes and hash bangs (backward compatablility)
-      url = href.replace(/^\//,'').replace('\#\!\/','')
+    event.preventDefault()
 
-      # Instruct Backbone to trigger routing events
-      app.router.navigate url, { trigger: true }
+    # Remove leading slashes and hash bangs (backward compatablility)
+    url = href.replace(/^\//,'').replace('\#\!\/','')
```
Comment (sderickson) [inline app/core/initialize.coffee]: What's this for exactly?

File: app/views/user/CertificatesView.coffee [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    window.print()
+
+  afterRender: ->
+    @autoSizeText '.student-name'
+
+  autoSizeText: (selector) ->
+    @$(selector).each (index, el) ->
+      while el.scrollWidth > el.offsetWidth or el.scrollHeight > el.offsetHeight
+        newFontSize = (parseFloat($(el).css('font-size').slice(0, -2)) * 0.95) + 'px'
+        $(el).css('font-size', newFontSize)
```
Comment (sderickson) [inline app/views/user/CertificatesView.coffee]: We need to get you acquainted with Vue

File: server/lib/bitly.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+config = require '../../server_config'
+BitlyClient = require('bitly')
+module.exports = BitlyClient(config.bitly.accessToken or '2ef4a986313451e8c44eeece891b80ed70af3398')  # TODO
```
Comment (sderickson) [inline server/lib/bitly.coffee]: What is this TODO?

File: app/core/initialize.coffee (hunk 1 above)
Comment (nwinter) [reply app/core/initialize.coffee]: `target="_blank"` was not working before. I fixed it and refactored this code while I was at it.

**Pull request merged**